src/contexts/auth.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { View, ActivityIndicator } from "react-native";
+import * as auth from '../services/Auth/auth'
+
+interface AuthContextData {
+  signed: boolean,
+  user: object | null,
+  loading: boolean,
+  signIn(email: string, password: string): Promise<void>,
+  signOut(): void
+}
+
+const AuthContext = createContext<AuthContextData>({} as AuthContextData)
+
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState<object | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
```

Comment:
suggestion: we could use inference of typescript

```suggestion
  const [loading, setLoading] = useState(false)
```